packages/block-library/src/navigation/editor.scss

```diff
@@ -1,23 +1,26 @@
+/**
+ * Editor only CSS.
+ */
+
 // Undo default editor styles.
-.editor-styles-wrapper .wp-block-navigation ul,
-.editor-styles-wrapper .wp-block-navigation ol {
-	margin-bottom: 0;
-	margin-left: 0;
-	padding-left: 0;
-}
+// These need extra specificity.
+.editor-styles-wrapper .wp-block-navigation {
+	ul {
```

Comment:
I removed the "ol" rule here, as it does not appear we use the ordered list for structure, anywhere in the code.